test/lib/jdk/test/lib/artifacts/Artifact.java

```diff
@@ -29,11 +29,18 @@
 @Repeatable(ArtifactContainer.class)
 @Retention(RetentionPolicy.RUNTIME)
 public @interface Artifact {
-    String organization();
-    String name();
-    String revision();
-    String extension();
+    String organization() default "";
+    String name() default "";
+    String revision() default "";
+    String extension() default "";
     String classifier() default "";
     boolean unpack() default true;
+
+    // For getting jdk versions
+    String server() default "";
+    String product() default "";
+    int version() default 0;
+    int build_number() default 0;
+    String file() default "";
```

Comment:
Are the changes in this file still needed? I think you were using Artifact and annotations in an earlier version of TestAutoCreateSharedArchiveUpgrade.java, but that would result in lots of duplications across {JDK versions} x { platforms}. That's why you chose to go with a lower-level API (the new ArtifactResolver.::resolve() method) so you can build the Map programmatically.